app/models/abstract_model.rb

```diff
@@ -350,11 +387,19 @@ def show_action
 
   # Return the URL of the "show_<object>" action
   #
+  #   # normalized controller
+  #   Article.index_action => "http://mushroomobserver.org/article/12"
+  #
+  #   # unnormalized controller
   #   Name.show_url(12) => "http://mushroomobserver.org/name/show_name/12"
   #   name.show_url     => "http://mushroomobserver.org/name/show_name/12"
   #
   def self.show_url(id)
-    "#{MO.http_domain}/#{show_controller}/#{show_action}/#{id}"
+    if controller_normalized?(name)
```

Comment:
And again.

Reply:
I changed the other ones, but left this as-is because
`return "#{MO.http_domain}/#{show_controller}/#{id}" if controller_normalized?(name)`
won't fit on one line (and would offend RuboCop).